python/paddle/fluid/dygraph/varbase_patch_methods.py

```diff
@@ -263,6 +263,36 @@ def __str__(self):
         from paddle.tensor.to_string import to_string
         return to_string(self)
 
+    def __deepcopy__(self, memo):
+        """
+        Deep copy Tensor, it will always performs Tensor copy.
+
+        Examples:
+            .. code-block:: python
+
+                import paddle
+                import copy
+                x = paddle.to_tensor(2.)
+                y = copy.deepcopy(x)
+                
+                print(x)
+                # Tensor(shape=[1], dtype=float32, place=CPUPlace, stop_gradient=True,
+                #        [2.])
+
+                print(y)
+                # Tensor(shape=[1], dtype=float32, place=CPUPlace, stop_gradient=True,
+                #        [2.])
+
+        """
+        if not self.is_leaf:
+            raise RuntimeError(
+                "Only Leaf Tensors support the deepcopy protocol at the moment, non-Leaf Tensors contains graph information that does't support deepcopy"
```

Comment:
`Tensors` -> `Tensor`?

deepcopy is enough, what is `deep copy protocol` ?

Reply:
python copy.py module will call .__reduce_ex__(4) to return current object 's __new__, __dict__.  4 is protocol, also have 0,1,2,3. User should not pay attention to it, deepcopy is enough,

Reply:
Done